python/tests/variant_tests/dtype_and_variant.py

```diff
@@ -0,0 +1,369 @@
+
+
+# FLOAT(REAL, DOUBLE)
+class varnttst_float_tbl(TstTable):
+    """Define the table used by the Float tests"""
+
+    def __init__(self):
+        self.sql = """CREATE TABLE varnt_float_tbl(
+                      real1 REAL,
+                      double1 DOUBLE
+                      )"""
+        self.data = [
+            {
+                "real1": 57681.18,
+                "double1": -38.2711234601246,
+            }
+        ]
+
+
+class varnttst_float_to_variant(TstView):
+    def __init__(self):
+        # checked manually
+        self.data = [{"real1_varnt": "57681.18", "double1_varnt": "-38.2711234601246"}]
+        self.sql = """CREATE MATERIALIZED VIEW float_to_variant AS SELECT
+                      CAST(real1 AS VARIANT) AS real1_varnt,
+                      CAST(double1 AS VARIANT) AS double1_varnt
+                      FROM varnt_float_tbl"""
+
+
+# Returns Decimal because: https://github.com/feldera/feldera/issues/3396
+class varnttst_variant_to_float(TstView):
+    def __init__(self):
+        # checked manually
+        self.data = [
+            {"real1": Decimal("57681.18"), "double1": Decimal("-38.2711234601246")}
+        ]
+        self.sql = """CREATE MATERIALIZED VIEW variant_to_float AS SELECT
+                      CAST(real1_varnt AS REAL) AS real1,
+                      CAST(double1_varnt AS DOUBLE) AS double1
+                      FROM float_to_variant"""
+
+
+# STRING(CHAR, CHAR(n), VARCHAR, VARCHAR(n)
+class varnttst_string_tbl(TstTable):
+    """Define the table used by the string tests"""
+
+    def __init__(self):
+        self.sql = """CREATE TABLE varnt_string_tbl(
+                      charr CHAR,
+                      charn CHAR(5),
+                      vchar VARCHAR,
+                      vcharn VARCHAR(5)
+                      )"""
+        self.data = [
+            {
+                "charr": "hi friend!",
+                "charn": "hello",
+                "vchar": "bye, mate!",
+                "vcharn": "ciao!",
+            }
+        ]
+
+
+# left to add CHAR: https://github.com/feldera/feldera/issues/3817
+class varnttst_string_to_variant(TstView):
+    def __init__(self):
+        # checked manually
+        self.data = [
+            {
+                "charn_varnt": '"hello"',
+                "vchar_varnt": '"bye, mate!"',
+                "vcharn_varnt": '"ciao!"',
+            }
+        ]
+        self.sql = """CREATE MATERIALIZED VIEW string_to_variant AS SELECT
+                      CAST(charn AS VARIANT) AS charn_varnt,
+                      CAST(vchar AS VARIANT) AS vchar_varnt,
+                      CAST(vcharn AS VARIANT) AS vcharn_varnt
+                      FROM varnt_string_tbl"""
+
+
+class varnttst_variant_to_string(TstView):
+    def __init__(self):
+        # checked manually
+        self.data = [{"charn": "hello", "vchar": "bye, mate!", "vcharn": "ciao!"}]
+        self.sql = """CREATE MATERIALIZED VIEW variant_to_string AS SELECT
```

Comment:
can you also make sure that long strings are truncated (we know this doesn't work for the table, but it should for the queries).